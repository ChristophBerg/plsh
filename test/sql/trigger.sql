\! mkdir /tmp/plsh-test && chmod a+rwx /tmp/plsh-test

CREATE FUNCTION shtrigger() RETURNS trigger AS $$
#!/bin/sh
(
for arg do
    echo "Arg is $arg"
done

printenv | LC_ALL=C sort | grep '^PLSH_TG_'
) >> /tmp/plsh-test/foo
chmod a+r /tmp/plsh-test/foo
exit 0
$$ LANGUAGE plsh;

CREATE TABLE pfoo (a int, b text);

CREATE TRIGGER testtrigger AFTER INSERT ON pfoo
    FOR EACH ROW EXECUTE PROCEDURE shtrigger('dummy');

CREATE TRIGGER testtrigger2 BEFORE UPDATE ON pfoo
    FOR EACH STATEMENT EXECUTE PROCEDURE shtrigger('dummy2');

INSERT INTO pfoo VALUES (1, 'one');
INSERT INTO pfoo VALUES (2, 'two');
INSERT INTO pfoo VALUES (3, 'three');

UPDATE pfoo SET b = 'oneone' WHERE a = 1;

\! cat /tmp/plsh-test/foo
\! rm -r /tmp/plsh-test
