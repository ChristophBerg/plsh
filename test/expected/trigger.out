\! mkdir /tmp/plsh-test && chmod a+rwx /tmp/plsh-test
CREATE FUNCTION shtrigger() RETURNS trigger AS '
#!/bin/sh
(
for arg do
    echo "Arg is $arg"
done
) >> /tmp/plsh-test/foo
chmod a+r /tmp/plsh-test/foo
exit 0
' LANGUAGE plsh;
CREATE TABLE pfoo (a int, b text);
CREATE TRIGGER testtrigger AFTER INSERT ON pfoo
    FOR EACH ROW EXECUTE PROCEDURE shtrigger('dummy');
CREATE TRIGGER testtrigger2 BEFORE UPDATE ON pfoo
    FOR EACH STATEMENT EXECUTE PROCEDURE shtrigger('dummy2');
INSERT INTO pfoo VALUES (1, 'one');
INSERT INTO pfoo VALUES (2, 'two');
INSERT INTO pfoo VALUES (3, 'three');
UPDATE pfoo SET b = 'oneone' WHERE a = 1;
\! cat /tmp/plsh-test/foo
Arg is dummy
Arg is 1
Arg is one
Arg is dummy
Arg is 2
Arg is two
Arg is dummy
Arg is 3
Arg is three
Arg is dummy2
\! rm -r /tmp/plsh-test
